lib/hydra/works/services/determine_mime_type.rb

```diff
@@ -0,0 +1,24 @@
+module Hydra::Works
+  class DetermineMimeType
+    # Determines the mime type for a given file
+    # @param [IO, File, Rack::Multipart::UploadedFile, #read] file
+    # @param [String, NilClass] original_name of the file
+    # @return [String]
+    class << self
+      def call(file, original_name = nil)
+        return file.mime_type if file.respond_to?(:mime_type) && file.mime_type.present?
+        return file.content_type if file.respond_to?(:content_type) && file.content_type.present?
+        mime_type_from_name_or_path(file, original_name) || default_mime_type
+      end
+
+      def default_mime_type
+        'application/octet-stream'
+      end
+
+      def mime_type_from_name_or_path(file, original_name)
+        return Hydra::PCDM::GetMimeTypeForFile.call(original_name) if original_name.present?
```

Comment:
@mjgiarlo it's going to relate to a forthcoming PR in projecthydra/curation_concerns@3cf65ca3352fad35d75e0f8505e623d4b45a0a6f that ultimately gets the original Sufia issue to pass. Do you see where I am here? Three layers down, and three gem releases just to fix one issue. 😉 

There are cases when we want to find the mime type using the provided original name, and _not_ the path. Examples are when my file has been uploaded as `sample.txt1234sdfasdf234`. The path doesn't help me there, so if the file is always represented as an object that responds to `original_name` or `original_filename`, I can supply the right name as opposed to having to rely on the path.